packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js

```diff
@@ -149,30 +149,47 @@ const getUtilityClass = (name) => {
 };
 
 const useSliderClasses = (props) => {
-  const { color, disabled, marked, orientation, track } = props;
+  const { color, disabled, marked, orientation, track, classes = {} } = props;
 
   const utilityClasses = {
-    root: clsx(getUtilityClass('root'), getUtilityClass(`color${capitalize(color)}`), {
-      'Mui-disabled': disabled,
-      [getUtilityClass('marked')]: marked,
-      [getUtilityClass('vertical')]: orientation === 'vertical',
-      [getUtilityClass('trackInverted')]: track === 'inverted',
-      [getUtilityClass('trackFalse')]: track === false,
-    }),
-    rail: getUtilityClass('rail'),
-    track: getUtilityClass('track'),
-    mark: getUtilityClass('mark'),
-    markLabel: getUtilityClass('markLabel'),
-    valueLabel: getUtilityClass('valueLabel'),
-    thumb: clsx(getUtilityClass('thumb'), getUtilityClass(`thumbColor${capitalize(color)}`), {
-      'Mui-disabled': disabled,
-    }),
+    root: clsx(
+      getUtilityClass('root'),
+      classes['root'],
+      getUtilityClass(`color${capitalize(color)}`),
+      classes[`color${capitalize(color)}`],
+      {
+        'Mui-disabled': disabled,
+        [getUtilityClass('marked')]: marked,
+        [classes['marked']]: marked,
+        [getUtilityClass('vertical')]: orientation === 'vertical',
+        [classes['vertical']]: orientation === 'vertical',
+        [getUtilityClass('trackInverted')]: track === 'inverted',
+        [classes['trackInverted']]: track === 'inverted',
+        [getUtilityClass('trackFalse')]: track === false,
+        [classes['trackFalse']]: track === false,
+      },
+    ),
+    rail: clsx(getUtilityClass('rail'), classes['rail']),
+    track: clsx(getUtilityClass('track'), classes['track']),
+    mark: clsx(getUtilityClass('mark'), classes['mark']),
+    markLabel: clsx(getUtilityClass('markLabel'), classes['markLabel']),
+    valueLabel: clsx(getUtilityClass('valueLabel'), classes['valueLabel']),
+    thumb: clsx(
+      getUtilityClass('thumb'),
+      classes['thumb'],
+      getUtilityClass(`thumbColor${capitalize(color)}`),
+      classes[`thumbColor${capitalize(color)}`],
+      {
+        'Mui-disabled': disabled,
+        [getUtilityClass('vertical')]: orientation === 'vertical',
+      },
+    ),
   };
 
   return utilityClasses;
 };
 
-const isComponent = (element) => typeof element !== 'string';
+const isHostComponent = (element) => typeof element === 'string';
```

Comment:
A note for the second component we will migrate, we might want to abstract this helper, we will need it everywhere

Reply:
Yeah, makes sense, probably there will be other things that can be abstracted, but for the thing I wasn't sure I just added them inline for now :)

Reply:
Yeah, better until we know for sure :)

Reply:
This discussion reminds me of something 🤔 

(something wet... 😉 )

Reply:
It was too fresh not to use it 😅